pyccel/stdlib/ndarrays/ndarrays.c

```diff
@@ -373,3 +429,65 @@ int64_t     *numpy_to_ndarray_shape(int64_t *np_shape, int nd)
     return nd_shape;
 
 }
+
+bool    is_same_shape(t_ndarray dest, t_ndarray src)
+{
+    int i = 0;
+
+    if (dest.nd != src.nd)
+        return false;
+    while (i < dest.nd)
+    {

```

Comment:
Is there a reason why you prefer `while` to `for`?

Reply:
It doesn't make a difference, but I find it is easier to miss the `++i` when using while loops. I would understand the use of while if you had:
```
while (i < dest.nd and dest.shape[i] == src.shape[i])
    ++i;
```
But in its current form I don't see any advantages